core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -494,22 +514,29 @@ public void createNamespace(
 
   @Override
   public List<Namespace> listNamespaces(SessionContext context, Namespace namespace) {
-    Map<String, String> queryParams;
-    if (namespace.isEmpty()) {
-      queryParams = ImmutableMap.of();
-    } else {
-      // query params should be unescaped
-      queryParams = ImmutableMap.of("parent", RESTUtil.NAMESPACE_JOINER.join(namespace.levels()));
+    Map<String, String> queryParams = Maps.newHashMap();
+    if (!namespace.isEmpty()) {
+      queryParams.put("parent", RESTUtil.NAMESPACE_JOINER.join(namespace.levels()));
     }
+    List<Namespace> namespaces = Lists.newArrayList();
+    String pageToken = "";
+    if (restPageSize != null) {
+      queryParams.put("pageSize", restPageSize);
+    }
```

Comment:
nit: the codebase usually adds a newline after the closing } from if statements

Reply:
will fix

Reply:
will fix